tests/php/ORM/DataQueryTest.php

```diff
@@ -215,28 +216,27 @@ public function testRelationOrderWithCustomJoin()
         $this->assertTrue(true);
     }
 
-    public function provideFieldCollision()
+    public static function provideFieldCollision()
     {
         return [
             'allow collisions' => [true],
             'disallow collisions' => [false],
         ];
     }
 
-    /**
-     * @dataProvider provideFieldCollision
-     */
+    #[DataProvider('provideFieldCollision')]
     public function testFieldCollision($allowCollisions)
     {
+        $this->enableErrorHandler();
         $dataQuery = new DataQuery(DataQueryTest\ObjectB::class);
         $dataQuery->selectField('COALESCE(NULL, 1) AS "Title"');
         $dataQuery->setAllowCollidingFieldStatements($allowCollisions);
 
         if ($allowCollisions) {
             $this->assertSQLContains('THEN "DataQueryTest_B"."Title" WHEN COALESCE(NULL, 1) AS "Title" IS NOT NULL THEN COALESCE(NULL, 1) AS "Title" ELSE NULL END AS "Title"', $dataQuery->sql());
         } else {
-            $this->expectError();
-            $this->expectErrorMessageMatches('/^Bad collision item /');
+            $this->expectException(ExpectedWarningException::class);
```

Comment:
While this previously had expectError(), running this code without $this->enableErrorHandler() will now throw a php warning, so expecting an ExepectedWarningException is correct